shared/login/signup/success/index.js

```diff
@@ -25,9 +25,10 @@ Success.propTypes = {
 }
 
 export default connect(
-  state => ({paperkey: state.signup.paperkey}),
+  (state, ownProps) => ({
+    paperkey: ownProps.paperkey || state.signup.paperkey,
```

Comment:
This is hacky -- we can either be being called from redux with entries in the store (which the signup flow prefers), or directly with `appendRouteElement()` (which the login flow prefers), so we try to support both.